### PR TITLE
Add  SHCB-1 support

### PR DESCRIPTION
**Pull request by tc-maxx**
I have now found the error, the entities already existed for me, so I didn't notice the error. I have now tested it again and here is the complete adjustment. Maybe you can implement it a little nicer ;) ? Thank you Max

### CODE REVIEW

File: python_scripts/shellies_discovery.py [hunk 1]
```diff
@@ -1,5 +1,5 @@
 """This script adds MQTT discovery support for Shellies devices."""
-VERSION = "4.5.4"
+VERSION = "4.5.4 - SHCB-1"
```
Comment (bieniu) [inline python_scripts/shellies_discovery.py:2]: Please do not edit this CONST in this PR.

File: python_scripts/shellies_discovery.py [hunk 2]
```diff
@@ -3230,6 +3268,70 @@ def mqtt_publish(topic, payload, retain, json=False):
...
+        KEY_STATE_TEMPLATE: "{%if value_json.ison%}on{%else%}off{%endif%}",
+        KEY_BRIGHTNESS_TEMPLATE: (
+            "{{value_json.brightness|float|multiply(2.55)|round(0)}}"
+        ),
+        KEY_COLOR_TEMP_TEMPLATE: (
+            "{% if value_json.ison and value_json.mode==^white^ %} {{ 1000000|multiply(1/(value_json.temp|float))|round(0) }} {%else%} 0 {%endif%}"
+        ),
+        KEY_RED_TEMPLATE: "{%if value_json.ison and value_json.mode==^color^ %} {{value_json.red}} {%else%} 0 {%endif%}",
+        KEY_GREEN_TEMPLATE: "{%if value_json.ison and value_json.mode==^color^ %} {{value_json.green}} {%else%} 0 {%endif%}",
+        KEY_BLUE_TEMPLATE: "{%if value_json.ison and value_json.mode==^color^ %} {{value_json.blue}} {%else%} 0 {%endif%}",
```
Comment (bieniu) [inline python_scripts/shellies_discovery.py:3322]:
```suggestion
        KEY_BLUE_TEMPLATE: "{%if value_json.ison and value_json.mode==^color^ %}{{value_json.blue}}{%else%}0{%endif%}",
```

**Review by bieniu — CHANGES_REQUESTED** [review]
The change works with my bulb. A few suggestions bellow.

**Comment by bieniu** [comment]
It seems there is a problem with effects:
```
2023-11-16 16:01:51.721 WARNING (MainThread) [homeassistant.components.mqtt.light.schema_template] Unsupported effect value received
2023-11-16 16:02:21.719 WARNING (MainThread) [homeassistant.components.mqtt.light.schema_template] Unsupported effect value received
2023-11-16 16:02:51.722 WARNING (MainThread) [homeassistant.components.mqtt.light.schema_template] Unsupported effect value received
```

**Comment by tc-maxx** [comment]
I haven't noticed the problem in the log yet. But the effect is set correctly. Apparently there are problems parsing the set effect.

There was still an error when setting the effect. Id:4 must be changed to id:3.

"{%elif effect==^Flash^%},^effect^:4" -> "{%elif effect==^Flash^%},^effect^:3"

File: python_scripts/shellies_discovery.py [hunk 3]
```diff
@@ -3230,6 +3266,70 @@ def mqtt_publish(topic, payload, retain, json=False):
...
+        KEY_MAX_MIREDS: light_options[KEY_MAX_MIREDS],
+        KEY_MIN_MIREDS: light_options[KEY_MIN_MIREDS],
+        KEY_UNIQUE_ID: unique_id,
+        KEY_QOS: str(qos),
+        KEY_DEVICE: device_info,
+        "~": default_topic,
+    }
+
+    mqtt_publish(config_topic, payload, retain, json=True)
+
```
Comment (bieniu) [inline python_scripts/shellies_discovery.py:3331]:
```suggestion
```

File: python_scripts/shellies_discovery.py [hunk 4]
```diff
@@ -3230,6 +3266,70 @@ def mqtt_publish(topic, payload, retain, json=False):
...
+            ",^mode^:^color^,^red^:{{red}},^green^:{{green}},^blue^:{{blue}}"
+            "{%endif%}"
+            "{%if brightness is defined%}"
+            ",^gain^:{{brightness|float|multiply(0.3922)|round}},"
+            "^brightness^:{{brightness|float|multiply(0.3922)|round}}"
+            "{%endif%}"
+            "{%if effect is defined%}"
+            "{%if effect==^Meteor Shower^%},^effect^:1"
+            "{%elif effect==^Gradual Change^%},^effect^:2"
+            "{%elif effect==^Flash^%},^effect^:4"
```
Comment (bieniu) [inline python_scripts/shellies_discovery.py:3300]: `4`? not `3`?

File: python_scripts/shellies_discovery.py [hunk 5]
```diff
@@ -3230,6 +3266,70 @@ def mqtt_publish(topic, payload, retain, json=False):
...
+        KEY_BRIGHTNESS_TEMPLATE: (
+            "{{value_json.brightness|float|multiply(2.55)|round(0)}}"
+        ),
+        KEY_COLOR_TEMP_TEMPLATE: (
+            "{%if value_json.ison and value_json.mode==^white^%}{{1000000|multiply(1/(value_json.temp|float))|round(0)}}{%else%}0{%endif%}"
+        ),
+        KEY_RED_TEMPLATE: "{%if value_json.ison and value_json.mode==^color^ %}{{value_json.red}}{%else%}0{%endif%}",
+        KEY_GREEN_TEMPLATE: "{%if value_json.ison and value_json.mode==^color^ %}{{value_json.green}}{%else%}0{%endif%}",
+        KEY_BLUE_TEMPLATE: "{%if value_json.ison and value_json.mode==^color^ %}{{value_json.blue}}{%else%}0{%endif%}",
+        KEY_EFFECT_TEMPLATE: "{{value_json.effect}}",
```
Comment (bieniu) [inline python_scripts/shellies_discovery.py:3321]: This looks wrong, I think we need to map the effect index to the effect name here.

File: python_scripts/shellies_discovery.py (hunk 4 above)
Comment (bieniu) [inline python_scripts/shellies_discovery.py:3300]:
```suggestion
            "{%elif effect==^Flash^%},^effect^:3"
```

File: python_scripts/shellies_discovery.py (hunk 5 above)
Comment (tc-maxx) [reply python_scripts/shellies_discovery.py:3321]: I tried around a bit and with the following payload I no longer get any errors:

`    payload = {
        KEY_SCHEMA: VALUE_TEMPLATE,
        KEY_NAME: light_name,
        KEY_COMMAND_TOPIC: light_options[KEY_COMMAND_TOPIC].format(light_id=light_id),
        KEY_STATE_TOPIC: light_options[KEY_STATE_TOPIC].format(light_id=light_id),
        KEY_AVAILABILITY: availability,
        KEY_COMMAND_ON_TEMPLATE: (
            "{^turn^:^on^"
            "{%if red is defined and green is defined and blue is defined%}"
            ",^mode^:^color^,^red^:{{red}},^green^:{{green}},^blue^:{{blue}}"
            "{%endif%}"
            "{%if brightness is defined%}"
            ",^gain^:{{brightness|float|multiply(0.3922)|round}},"
            "^brightness^:{{brightness|float|multiply(0.3922)|round}}"
            "{%endif%}"
            "{%if effect is defined%}"
            "{%if effect==^Meteor Shower^%},^effect^:1"
            "{%elif effect==^Gradual Change^%},^effect^:2"
            "{%elif effect==^Flash^%},^effect^:3"
            "{%else%},^effect^:0"
            "{%endif%}"
            "{%else%},^effect^:0"
            "{%endif%}"
            "{%if color_temp is defined%},^mode^:^white^,"
            "^temp^:{{(1/(color_temp|float))|multiply(1000000)|round}}"
            "{%endif%}}"
        ),
        KEY_COMMAND_OFF_TEMPLATE: "{^effect^:0,^turn^:^off^}",
        KEY_STATE_TEMPLATE: "{%if value_json.ison%}on{%else%}off{%endif%}",
        KEY_BRIGHTNESS_TEMPLATE: "{%if value_json.ison and value_json.mode==^white^%}{{value_json.brightness|float|multiply(2.55)|round(0)}}{%else%}{{value_json.gain|float|multiply(2.55)|round(0)}}{%endif%}",
        KEY_RED_TEMPLATE: "{%if value_json.ison and value_json.mode==^color^%}{{value_json.red}}{%else%}0{%endif%}",
        KEY_GREEN_TEMPLATE: "{%if value_json.ison and value_json.mode==^color^%}{{value_json.green}}{%else%}0{%endif%}",
        KEY_BLUE_TEMPLATE: "{%if value_json.ison and value_json.mode==^color^%}{{value_json.blue}}{%else%}0{%endif%}",
        KEY_EFFECT_LIST: ["Aus", "Meteor Shower", "Gradual Change", "Flash"],
        KEY_EFFECT_STATE_TOPIC: light_options[KEY_STATE_TOPIC].format(light_id=light_id),
        KEY_EFFECT_VALUE_TEMPLATE: "{%if value_json.effect==1%}Meteor Shower{%elif value_json.effect==2%}Gradual Change{%elif value_json.effect==3%}Flash{%else%}Aus{%endif%}",
        KEY_COLOR_TEMP_TEMPLATE: "{%if value_json.ison and value_json.mode==^white^%}{{1000000|multiply(1/(value_json.temp|float))|round(0)}}{%else%}0{%endif%}",
        KEY_MAX_MIREDS: light_options[KEY_MAX_MIREDS],
        KEY_MIN_MIREDS: light_options[KEY_MIN_MIREDS],
        KEY_UNIQUE_ID: unique_id,
        KEY_QOS: str(qos),
        KEY_DEVICE: device_info,
        "~": default_topic,
    }`

File: python_scripts/shellies_discovery.py [hunk 6]
```diff
@@ -3230,6 +3268,65 @@ def mqtt_publish(topic, payload, retain, json=False):
...
+            "^temp^:{{(1/(color_temp|float))|multiply(1000000)|round}}"
+            "{%endif%}}"
+        ),
+        KEY_COMMAND_OFF_TEMPLATE: "{^effect^:0,^turn^:^off^}",
+        KEY_STATE_TEMPLATE: "{%if value_json.ison%}on{%else%}off{%endif%}",
+        KEY_BRIGHTNESS_TEMPLATE: "{%if value_json.ison and value_json.mode==^white^%}{{value_json.brightness|float|multiply(2.55)|round(0)}}{%else%}{{value_json.gain|float|multiply(2.55)|round(0)}}{%endif%}",
+        KEY_RED_TEMPLATE: "{%if value_json.ison and value_json.mode==^color^%}{{value_json.red}}{%else%}0{%endif%}",
+        KEY_GREEN_TEMPLATE: "{%if value_json.ison and value_json.mode==^color^%}{{value_json.green}}{%else%}0{%endif%}",
+        KEY_BLUE_TEMPLATE: "{%if value_json.ison and value_json.mode==^color^%}{{value_json.blue}}{%else%}0{%endif%}",
+        KEY_EFFECT_LIST: ["Aus", "Meteor Shower", "Gradual Change", "Flash"],
```
Comment (bieniu) [inline python_scripts/shellies_discovery.py:3316]: `"Aus"` - Please use English.

File: python_scripts/shellies_discovery.py [hunk 7]
```diff
@@ -3230,6 +3268,65 @@ def mqtt_publish(topic, payload, retain, json=False):
...
+        KEY_EFFECT_STATE_TOPIC: light_options[KEY_STATE_TOPIC].format(light_id=light_id),
+        KEY_EFFECT_VALUE_TEMPLATE: "{%if value_json.effect==1%}Meteor Shower{%elif value_json.effect==2%}Gradual Change{%elif value_json.effect==3%}Flash{%else%}Aus{%endif%}",
```
Comment (bieniu) [inline python_scripts/shellies_discovery.py:3318]: `effect_value_template` is not a part of the template schema https://www.home-assistant.io/integrations/light.mqtt/#template-schema

File: python_scripts/shellies_discovery.py [hunk 8]
```diff
@@ -3230,6 +3268,65 @@ def mqtt_publish(topic, payload, retain, json=False):
...
+            "{%endif%}}"
+        ),
+        KEY_COMMAND_OFF_TEMPLATE: "{^effect^:0,^turn^:^off^}",
+        KEY_STATE_TEMPLATE: "{%if value_json.ison%}on{%else%}off{%endif%}",
+        KEY_BRIGHTNESS_TEMPLATE: "{%if value_json.ison and value_json.mode==^white^%}{{value_json.brightness|float|multiply(2.55)|round(0)}}{%else%}{{value_json.gain|float|multiply(2.55)|round(0)}}{%endif%}",
+        KEY_RED_TEMPLATE: "{%if value_json.ison and value_json.mode==^color^%}{{value_json.red}}{%else%}0{%endif%}",
+        KEY_GREEN_TEMPLATE: "{%if value_json.ison and value_json.mode==^color^%}{{value_json.green}}{%else%}0{%endif%}",
+        KEY_BLUE_TEMPLATE: "{%if value_json.ison and value_json.mode==^color^%}{{value_json.blue}}{%else%}0{%endif%}",
+        KEY_EFFECT_LIST: ["Aus", "Meteor Shower", "Gradual Change", "Flash"],
+        KEY_EFFECT_STATE_TOPIC: light_options[KEY_STATE_TOPIC].format(light_id=light_id),
```
Comment (bieniu) [inline python_scripts/shellies_discovery.py:3317]: `effect_state_topic` is not a part of the template schema https://www.home-assistant.io/integrations/light.mqtt/#template-schema

**Review by bieniu — CHANGES_REQUESTED** [review]

File: python_scripts/shellies_discovery.py (hunk 7 above)
Comment (tc-maxx) [reply python_scripts/shellies_discovery.py:3318]: Where did you get the error message from?
You can find `effect_value_template` in the schema here:

https://www.home-assistant.io/integrations/light.mqtt/#effect_value_template

File: python_scripts/shellies_discovery.py (hunk 8 above)
Comment (tc-maxx) [reply python_scripts/shellies_discovery.py:3317]: [effect_state_topic ](https://www.home-assistant.io/integrations/light.mqtt/#effect_state_topic)

Comment (bieniu) [reply python_scripts/shellies_discovery.py:3317]: `effect_state_topic` is part of  the default schema.

**Comment by bieniu** [comment]
Are you going to finish this PR? I think we were very close to proper implementation.

**Pull request closed without merging**